client/dashboard/sites/settings-static-file-404/index.tsx

```diff
@@ -0,0 +1,164 @@
+import { DataForm } from '@automattic/dataviews';
+import { useMutation, useQuery } from '@tanstack/react-query';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalVStack as VStack,
+	Card,
+	CardBody,
+	Button,
+} from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { settings } from '@wordpress/icons';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState } from 'react';
+import { siteQuery, siteStaticFile404Query, siteStaticFile404Mutation } from '../../app/queries';
+import { Callout } from '../../components/callout';
+import PageLayout from '../../components/page-layout';
+import SettingsPageHeader from '../settings-page-header';
+import calloutIllustrationUrl from './callout-illustration.svg';
+import type { Site } from '../../data/types';
+import type { Field } from '@automattic/dataviews';
+
+export function canSetStaticFile404Handling( site: Site ) {
+	return site.is_wpcom_atomic;
+}
+
+const fields: Field< { setting: string } >[] = [
+	{
+		id: 'setting',
+		label: __( 'Server response' ),
+		Edit: 'select',
+		description: __(
+			'Assets are images, fonts, JavaScript, and CSS files that web browsers request as part of loading a web page. This setting controls how the web server handles requests for missing asset files.'
+		),
+		elements: [
+			{ value: 'default', label: __( 'Default' ) },
+			{ value: 'lightweight', label: __( 'Send a lightweight File-Not-Found page' ) },
+			{ value: 'wordpress', label: __( 'Delegate request to WordPress' ) },
+		],
+	},
+];
+
+const form = {
+	type: 'regular' as const,
+	fields: [ 'setting' ],
+};
+
+const getPageLayoutProps = () => ( {
+	size: 'small',
+	header: <SettingsPageHeader title={ __( 'Handling requests for nonexistent assets' ) } />,
+} );
+
+export default function SiteStaticFile404Settings( { siteSlug }: { siteSlug: string } ) {
+	const { createSuccessNotice, createErrorNotice } = useDispatch( noticesStore );
+	const { data: site } = useQuery( siteQuery( siteSlug ) );
+	const { data: currentSetting } = useQuery( {
+		...siteStaticFile404Query( siteSlug ),
+		enabled: site && canSetStaticFile404Handling( site ),
+	} );
+	const mutation = useMutation( siteStaticFile404Mutation( siteSlug ) );
+
+	const [ formData, setFormData ] = useState< { setting: string } >( {
+		setting: currentSetting ?? 'default',
+	} );
+
+	if ( ! site ) {
+		return null;
+	}
+
+	if ( ! canSetStaticFile404Handling( site ) ) {
+		const handleUpgradePlan = () => {
+			const params = new URLSearchParams();
+			const backUrl = window.location.href.replace( window.location.origin, '' );
+
+			params.append( 'cancel_to', backUrl );
+			params.append( 'redirect_to', backUrl );
+
+			window.location.href = `/checkout/${ encodeURIComponent(
+				siteSlug
+			) }/business?${ params.toString() }`;
+		};
+
+		return (
+			<PageLayout { ...getPageLayoutProps() }>
+				<Callout
```

Comment:
(not a blocker) I think this callout is meant to be shared for the other server setting upsell pages as well? Might worth extracting this, we can do later.

Reply:
I can follow up this next together with the upsells in PHP and Database 🙂